subsys/power/power.c

```diff
@@ -208,12 +208,6 @@ static enum pm_state pm_policy_mgr(int32_t ticks)
 	return z_power_state.state;
 }
 
-
-enum pm_state pm_system_suspend(int32_t ticks)
```

Comment:
this was added by me as a placeholder, idle thread is supposed to call the the policy manager  and remove `pm_system_suspend` which is misleading. `pm_system_suspend` being called in `idle()` gives the impression that we are going to suspend everytime when in fact, the policy manager decided which state we transition to.

So I would remove the `pm_system_suspend()` from `idle()` and call policy manager directly and adapt in other places as well.

Reply:
that is fine, and makes the other commit not necessary. Let me re-work this.

Reply:
actually thinking a little bit more about it we have two possibilities here:
1) idle thread -> policy manager  -> power subsystem
2) idle thread -> power subsystem -> policy manager

We are currently doing 2 but taking the approach 1 may simplify some things, let me change it to check

Reply:
Hey @nashif I have looked it more carefully and there are some drawbacks making the idle thread talk directly with the policy. First, pm_system_suspend will need be public, because all policies will need to call it, and our model allows application create their own policies. This api is not intend to be public at all. The other reason is that I think we want to keep the policy simple, just returning the next state that the system should go and the power subsystem is responsible for the rest. We can discuss it, but I prefer discuss it before do the changes.

Regarding the header, the problem with creating a `power_internal.h` is that this header is not going to be use in the power subsystem but in the kernel. I didn't see other subsystems doing it. Don't know how is the best way to organize it. The situation is that we have a functionality that is implemented by this subsystem that is only used by the kernel, not even the subsystem itself needs to call it, that is similar of `arch_interface.h` but for this subsystem.

Anyway, I sent a new version fixing the commit message and adding more cleanup, please review it again and let me know about the header. The `pm_system_suspend` call from the idle thread is something we can discuss more, this commit is not introducing it there, just simplifying.